frontend/packages/dev-console/src/components/monitoring/alerts/SilenceDurationDropdown.tsx

```diff
@@ -37,29 +48,58 @@ const SilenceDurationDropDown: React.FC<SilenceDurationDropDownProps> = ({ rule
   ];
 
   const setDuration = (duration: string) => {
-    if (duration !== 'silenceFor') {
-      const startsAt = new Date();
-      const endsAt = new Date(startsAt.getTime() + parsePrometheusDuration(duration));
+    const startsAt = new Date();
+    const endsAt = new Date(startsAt.getTime() + parsePrometheusDuration(duration));
 
-      const payload = {
-        createdBy,
-        endsAt: endsAt.toISOString(),
-        startsAt: startsAt.toISOString(),
-        matchers,
-        comment: '',
-      };
-
-      coFetchJSON.post(`${ALERT_MANAGER_TENANCY_BASE_PATH}/api/v2/silences`, payload);
-    }
+    const payload = {
+      createdBy,
+      endsAt: endsAt.toISOString(),
+      startsAt: startsAt.toISOString(),
+      matchers,
+      comment: '',
+    };
+    setSilencing(true);
+    coFetchJSON
+      .post(`${ALERT_MANAGER_TENANCY_BASE_PATH}/api/v2/silences`, payload)
```

Comment:
Same comment as for the `delete`. In the admin Console, after creating a silence, we call `refreshNotificationPollers` to trigger a refresh of the alert's `state`, so I think we could do the same here.

Reply:
Our scenario is a little different. In our case, we need to trigger the refresh for rules and silences. But in `refreshNotificationPollers` it is refreshing only silences.

Reply:
I looked into this some more and I still think we can simplify the code here by using `refreshNotificationPollers` or a similar approach. However, I'm OK with addressing that with a follow up PR.